### ISSUE TITLE
Duplicate context menu entry, one greyed out - after changing installlocation (ProgramData -> ProgramFiles) architecture (x86 -> x64) and version (7.9.5 -> 8.5.4)

### ISSUE THREAD

**Issue by Tirakk** [notepad-plus-plus/notepad-plus-plus#13871]
### Description of the Issue
We are trying to update Notepad++ in an Enterprise Environment from version 7.9.5 **x86** to version 8.5.4 **x64**
We formerly installed Notepad++ 7.9.5 x86 with parameters "/S /D=C:\ProgramData" because of reasons, but want to change the install directory of 8.5.4 to the default "C:\Program Files".
Therefor we are uninstalling Notepad++ 7.9.5 with "C:\ProgramData\Notepad++\uninstall.exe /S" with Microsoft's Configuration Manager as the "nt-authority\system" user - first inconvenience is, that "NppShell_06.dll" is locked and wont be deleted on uninstall.
Secondly, when we install Notepad++ with parameter "/S", initially the context menu (classic - we disabled the modern context menu in the registry) entry doesn't appear. On reboot or logout/login the context menu entry is created (in english, system language is german) - but when we open the context menu a second time, a second entry appears (both in german now) - greyed out and not working:

After uninstallation and installation:
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/77411182/8c5c0110-9495-4a5a-9f57-82253c35878a)

After logout and login:
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/77411182/081afc1e-19e2-4c40-bca2-42551796550b)

After opening the context menu the second time:
![image](https://github.com/notepad-plus-plus/notepad-plus-plus/assets/77411182/0e68638d-38c5-4a77-932b-8ae5a8d4683d)

### Steps to Reproduce the Issue
1. Disable modern context menu: reg.exe add "HKCU\Software\Classes\CLSID\{86ca1aa0-34aa-4e8b-a509-50c905bae2a2}\InprocServer32" /f /ve
2. Install Notepad++ 7.9.5 x86 in system context with parameters "/S /D=C:\Programdata"
3. Uninstall Notepad++ 7.9.5 x86 in system context with parameters "/S"
4. Install Notepad++ 8.5.4 x64 in system context with parameters "/S"
5. Open context menu on file: No "Edit with Notepad++" entry
6. Relog into the OS and open context menu on a file again: "Edit with Notepad++" entry appears
7. Reopen context menu: a second entry appears and the language ist changed.

### Expected Behavior
After installation there should be one context menu entry.

### Actual Behavior
Duplicate context menu entry - one of them greyed out and not working.

### Debug Information
Notepad++ v8.5.4   (64-bit)
Build time : Jun 17 2023 - 20:42:45
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : "C:\Program Files (x86)\netinst\NIInst32.exe" 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 11 Enterprise (64-bit)
OS Version : 21H2
OS Build : 22000.2057
Current ANSI codepage : 1252
Plugins : 
    ComparePlugin (2.0.2)
    DSpellCheck (1.5)
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)

**Comment by Tirakk** [comment]
Even after a manual uninstall and new install, the problem persists.
I even tried updating with the buildin Update mechanism - the same problem with a duplicate context menu entry occurs.

Does noone else have this problem?

**Comment by jvoigtlaender** [comment]
Oh yes, you are not the only one.

**Comment by sombraguerrero** [comment]
I don't pretend to be masterful enough of the registry to understand where the true redundancy is here, but the problem is clearly in the installer's logic when it's writing the CLSID entries. (UUID pulled from the installer's manifest file for 64-bit 8.5.6).

```
Computer\HKEY_CLASSES_ROOT\PackagedCom\ClassIndex\{E6950302-61F0-4FEB-97DB-855E30D4A991}

Computer\HKEY_CLASSES_ROOT\PackagedCom\Package\NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6\Class\{E6950302-61F0-4FEB-97DB-855E30D4A991}

Computer\HKEY_CLASSES_ROOT\PackagedCom\Package\NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6\Server\0

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Classes\PackagedCom\ClassIndex\{E6950302-61F0-4FEB-97DB-855E30D4A991}

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Classes\PackagedCom\Package\NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6\Class\{E6950302-61F0-4FEB-97DB-855E30D4A991}

Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Classes\PackagedCom\Package\NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6\Server\0
---------------------------------------------------------------------------------------------------------
Computer\HKEY_CURRENT_USER\Software\Microsoft\Windows\CurrentVersion\Shell Extensions\Cached

HKEY_USERS\S-1-5-21-1627273618-842426610-1179000955-16308\Software\Microsoft\Windows\CurrentVersion\Shell Extensions\Cached
```

**Comment by sombraguerrero** [comment]
> From the release notes of Notepad++ version 8.5.2:

Enhance “Edit with Notepad++” context menu and fix several issues. (Fix [#13399 5](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13399), [#13410 1](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13410), [#13425 1](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13425), [#13438 1](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13438), [#13440 1](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13440), [#13446 4](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13446), [#13457 2](https://github.com/notepad-plus-plus/notepad-plus-plus/issues/13457))

So I would guess that the following 2 commands (in order) should bring things back to normal:

regsvr32.exe /u "c:\path to\NppShell.dll"
regsvr32.exe "c:\path to\NppShell.dll"
This would remove the context menu entries and reinstall them.

(I wrote my own Notepad++ context menu entries, but a quick test with 8.5.2 installs just 1 extra "Edit with Notepad++" entry. Which is how it should be).

**Comment by sombraguerrero** [comment]
> Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Classes\PackagedCom\Package\NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6\Server\0
I think this one is probably the best one to delete for the desired effect.

**Comment by Tirakk** [comment]
> > Computer\HKEY_LOCAL_MACHINE\SOFTWARE\Classes\PackagedCom\Package\NotepadPlusPlus_1.0.0.0_neutral__7njy0v32s6xk6\Server\0
> 
> Based on my testing, I think this one is probably the best one to delete for the desired effect.

Thank you for sharing your testing and work on this issue.
After some testing I found this key is only created after restart of the Explorer process and after opening the context menu for the first time - so I do not see any possibility to remove the key directly after the install in system context.

I think for now I will resort to deleting the mentioned key via SCCM Configuration Item - this works for now.

**Comment by sombraguerrero** [comment]
Yeah, my impression based on other discussion I've read through is that this is actually somewhat deliberate because one of these entries is meant to accommodate Windows 11's "new" context menu layout, which is for some reason handled differently in the registry than the "old" layout.
